modules/generator/processor/localblocks/query_range.go

```diff
@@ -101,17 +111,18 @@ func (p *Processor) QueryRange(ctx context.Context, req *tempopb.QueryRangeReque
 			continue
 		}
 
-		wg.Add(1)
-		go func(b *ingester.LocalBlock) {
-			defer wg.Done()
-			resp, err := p.queryRangeCompleteBlock(ctx, b, *req, timeOverlapCutoff, unsafe, int(req.Exemplars))
-			if err != nil {
-				jobErr.Store(err)
-				return
-			}
-
-			jobEval.ObserveSeries(resp)
-		}(b)
+		if maxSeries == 0 || (int(syncAtomic.LoadInt32(&totalRawResultsSeriesCount))+jobEval.Length() < maxSeries) {
```

Comment:
i don't think this check is correct. it assumes that all series in "jobEval" are distinct from all series in "eval".